modules/advertlyBidAdapter.js

```diff
@@ -0,0 +1,129 @@
+import { registerBidder } from '../src/adapters/bidderFactory';
+import { config } from '../src/config';
+import * as utils from '../src/utils';
+import {BANNER, VIDEO} from '../src/mediaTypes';
+import { ajax } from '../src/ajax';
+import {Renderer} from '../src/Renderer';
+
+const SUPPORTED_AD_TYPES = [BANNER, VIDEO];
+const BIDDER_CODE = 'advertly';
+const DOMAIN = 'http://api.advertly.com/';
+const RENDERER_URL = '//acdn.adnxs.com/video/outstream/ANOutstreamVideo.js';
+
+function isBidRequestValid(bid) {
+  return (typeof bid.params !== 'undefined' && parseInt(utils.getValue(bid.params, 'publisherId')) > 0);
+}
+
+function buildRequests(validBidRequests) {
+  return {
+    method: 'POST',
+    url: DOMAIN + 'www/admin/plugins/Prebid/getAd.php',
+    options: {
+      withCredentials: false,
+      crossOrigin: true
+    },
+    data: validBidRequests,
+  };
+}
+
+function interpretResponse(serverResponse, request) {
+  const response = serverResponse.body;
+  const bidResponses = [];
+  var bidRequestResponses = [];
+
+  utils._each(response, function(bidAd) {
+    bidAd.adResponse = {
```

Comment:
Create new objects instead of modifying response data

Reply:
New object created

Reply:
That won't create new objects. `bnd` points to the same `bidAd` object as before.

`Object.assign(bnd, bidAd)` would create a shallow copy if the entire `bidAd` is needed. Otherwise, you could just set individual fields on `bnd`.

Also, `let bnd` should be within the loop instead of outside.

Reply:
Above changes are done

Reply:
Hi @msm0504 ,
We have updated the new objects like you comment.

Reply:
`bnd = bidAd;` (line 36) is no longer needed

Reply:
We have removed bnd = bidAd; (line 36).